src/lib/utils/expression.ts

```diff
@@ -0,0 +1,25 @@
+import type { ExpressionSpecification } from 'maplibre-gl';
+
+import { COLORS, COLOR_ORDER } from './constants';
+
+/**
+ * Create a Maplibre GL JS match expression for a given field and categories.
+ *
+ * @param field – The field to match against, one of 'Agency Name', 'Category',
+ * or 'Funding Source'.
+ * @param categories – The categories to match against.
+ * @returns – A Maplibre GL JS match expression.
+ */
```

Comment:
Here's an example JSDoc comment for a function. In general, I really like to use this convention for documenting functions that live in "utils" style folders. Better yet, this doc string will appear on hover at call sites in the codebase in an IDE like VSCode.